testutil/test_app.go

```diff
@@ -163,10 +197,103 @@ func generateKeyring(t *testing.T, accts ...string) keyring.Keyring {
 	return kb
 }
 
+// SetupWithGenesisValSet initializes GenesisState with a single validator and genesis accounts
+// that also act as delegators.
+func GenesisStateWithSingleValidator(t *testing.T, testApp *app.App) (app.GenesisState, *tmtypes.ValidatorSet) {
```

Comment:
This might fix the issue @renaynay was running into with using a test application in celestia-node, as the network could not be setup due to improper initialization of the staking state/validator set